src/chef_authn.erl

```diff
@@ -115,8 +116,16 @@ process_key({'Certificate', _Der, _} = CertEntry) ->
     TbsCert = Cert#'Certificate'.tbsCertificate,
     Spki = TbsCert#'TBSCertificate'.subjectPublicKeyInfo,
     {0, KeyDer} = Spki#'SubjectPublicKeyInfo'.subjectPublicKey,
-    public_key:der_decode('RSAPublicKey', KeyDer).
-
+    public_key:der_decode('RSAPublicKey', KeyDer);
+process_key({'PrivateKeyInfo', _, _} = Entry) ->
+    KeyInfo = public_key:pem_entry_decode(Entry),
+    KeyAlgorithmInfo = KeyInfo#'PrivateKeyInfo'.privateKeyAlgorithm,
+    #'PrivateKeyInfo_privateKeyAlgorithm'{algorithm=Algorithm} = KeyAlgorithmInfo,
+    case Algorithm of
+        ?'rsaEncryption' ->
+            PrivateKey = KeyInfo#'PrivateKeyInfo'.privateKey,
+            public_key:der_decode('RSAPrivateKey', list_to_binary(PrivateKey))
+    end.
```

Comment:
I used the case statement in case we ever needed to add another algorithm. Also, if it does not match `?'rsaEncryption'`, it errors out.